### PR TITLE
docs: add migration guide for users upgrading from dual-binary versions

### PR DESCRIPTION
**Pull request by behrangsa**
## Summary

Adds a comprehensive migration guide for users upgrading from dual-binary versions of Samoyed.

**Addresses issue #78 (AC 63.7)** - Migration guide created for users upgrading from dual-binary versions

## Changes Made

 **Migration Guide Section**: Added to README.md positioned prominently between Installation and Quick Start

 **Clear Before/After Comparison**:
- Before: `exec samoyed-hook "pre-commit" "$@"`
- After: `exec samoyed hook "pre-commit" "$@"`

 **Step-by-Step Migration Instructions**:
1. Update Samoyed installation (`cargo install samoyed`)
2. Re-initialize hooks (`samoyed init` - safe to run multiple times)
3. Verify migration success (`cat .samoyed/_/pre-commit`)

 **Deprecation Timeline**:
- **Now**: `samoyed-hook` binary works but shows deprecation warnings
- **September 1, 2025**: `samoyed-hook` binary will be removed entirely
- **Recommendation**: Migrate immediately to avoid future issues

 **Troubleshooting Section**: Common issues and solutions for migration problems

### User-Friendly Approach
- Positioned prominently in README so users see it immediately after installation
- Emphasizes that `samoyed init` is safe to run multiple times
- Clear verification steps to confirm successful migration

### Timeline Clarity
- Explicit deprecation timeline with specific removal date
- Graduated approach: warnings now, removal later
- Clear recommendation to migrate immediately

### Safety Focus
- Re-initialization is non-destructive and can be repeated
- Verification steps to ensure migration worked correctly
- Troubleshooting section for edge cases

## Testing

-  All 258 tests pass after adding migration guide
-  No functional code changes, only documentation
-  Hook execution continues to work during migration testing

## Type of Change

 Documentation update

## Impact

-  Users upgrading from v0.1.x will have clear migration path
-  Prevents confusion during architecture transition period  
-  Reduces support burden with proactive guidance
-  Sets clear expectations for deprecation timeline

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,6 +29,51 @@ Samoyed is published on [crates.io](https://crates.io/crates/samoyed):
...
+
+1. **Update Samoyed**: Install the latest version
+   ```bash
+   cargo install samoyed
+   ```
+
+2. **Re-initialize Your Hooks**: This updates your `.samoyed/_/*` files to use the new unified binary
+   ```bash
+   # In your project directory
+   samoyed init
```
[the remark this answers is not in the document]
Comment (behrangsa) [reply README.md:54]:  **Updated!** The migration guide now includes both initialization options with clear guidance on when to use each.

**Commit:** c9d86f7

**Updated Migration Steps:**
```bash
# Option 1: Standard re-initialization (recommended)
# Safe to run multiple times, updates hooks only if needed
samoyed init

# Option 2: Force re-initialization (explicit migration)
# Explicitly overwrites all hook files to ensure complete migration
samoyed init -f _
```

**When to use each option:**
- Use `samoyed init` for normal migration (safe, non-destructive)
- Use `samoyed init -f _` if you want to force complete re-creation of all hook files

This addresses your feedback about the `-f` flag and provides users with both the safe default and the explicit forced option for migration scenarios.
